public-api/src/main/java/org/pmiops/workbench/publicapi/DataBrowserController.java

```diff
@@ -452,6 +456,9 @@ public ConceptAnalysis apply(ConceptAnalysis ca) {
         // Too slow and concept names wrong so we hardcode list
         // List<Concept> genders = conceptDao.findByConceptClassId("Gender");
 
+        //Get the list of questions that has sub questions in survey
+        List<Long> hasSubQuestions = questionConceptDao.findSurveyMainQuestionIds(Long.valueOf(surveyConceptId)).stream().map(QuestionConcept::getConceptId).collect(Collectors.toList());
```

Comment:
Should collect this into a Set instead - since you're doing O(n) lookups on L478 otherwise

Reply:
Yes, changed it to set. tahnks.